src/shared/components/Button.tsx

```diff
@@ -0,0 +1,31 @@
+import { MouseEvent, ReactNode } from "react";
+import styles from "./Button.module.css";
+
+interface Props {
+  className?: string;
+  type: "button" | "submit" | "reset";
+  disabled?: boolean;
+  children?: ReactNode;
+  onClick?: (e: MouseEvent<HTMLButtonElement>) => void;
+}
+
+const Button = ({
+  type = "button",
+  className = "",
+  disabled = false,
+  children,
+  onClick = () => {},
+}: Props) => {
```

Comment:
P1:
아래 타입들은 버튼 태그의 속성과 동일하므로 아래처럼 작성하시는 것을 추천드립니다.
children속성을 받을때는 react의 PropsWithChildren 타입을 활용하셔도 좋습니다.
이렇게 작성하시면 버튼 태그의 속성값들을 다 타입에 추가하지 않으셔도 사용이 가능합니다.

```suggestion
interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
 /** 추가 타입 정의가 필요할 경우 여기에 */
}

const Button = ({
  type = "button",
  className = "",
  disabled = false,
  children,
  onClick = () => {},
}: PropsWithChildren<Props>) => {
```